### ISSUE TITLE
user creation with SCIM

### ISSUE THREAD

**Issue by aswincoolsaet** [directus/docs#154]
user creation through SCIM fails, because of the validator:

password: This value should not be blank.
timezone: This value should not be blank.
locale: This value should not be blank.
roles: This value should not be blank.

according to documentation these values are not needed. 
https://docs.directus.io/api/reference.html#create-scim-user

**Comment by aswincoolsaet** [comment]
Web Response: {"schemas":["urn:ietf:params:scim:api:messages:2.0:Error"],"status":422,"detail":"password: This value should not be blank. timezone: This value should not be blank. locale: This value should not be blank. roles: This value should not be blank."}.

**Comment by benhaynes** [comment]
@bjgajjar — should we plan on updating the Docs or the API? I assume this is an issue of Directus requiring that info even though SCIM doesn't?

**Comment by verbman** [comment]
I want to assign roles via an action filter which I was doing, I don't want to give 'create' access to directus_user_roles. However when I post a user I get the error: 'roles: This value should not be blank' - and if I add the role to my post `roles: [{role: '2'}]` I get the permission error 'Creating item to \"directus_user_roles\" collection was denied'. 

This leaves no option but 'create' access on directus_user_roles, a filter to make it safe (i.e. set to public), and then a script to safely update it afterwards  to the desired role. This is no ideal.

In my mind therefore 'role' should allowed to be blank.

**Comment by benhaynes** [comment]
Yeah, what you're describing makes sense. I'm not sure the best way to approach this though.

We have Directus User fields that are required by the platform, but that means that we need that info when a user is created.

One potential solution would be an unrelated idea I had: that required fields wouldn't be required for certain status options. Then you could add a user (in draft mode) and it wouldn't require/error on the missing fields.

Would that maybe be a work-around for this?
